presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetCompressionUtils.java

```diff
@@ -86,30 +89,25 @@ private static Slice decompressGzip(Slice input, int uncompressedSize)
         }
     }
 
+    // decompress LZO-compressed data using HADOOP-LZO codec
     private static Slice decompressLZO(Slice input, int uncompressedSize)
+            throws IOException
     {
-        LzoDecompressor lzoDecompressor = new LzoDecompressor();
-        long totalDecompressedCount = 0;
-        // over allocate buffer which makes decompression easier
-        byte[] output = new byte[uncompressedSize + SIZE_OF_LONG];
-        int outputOffset = 0;
-        int inputOffset = 0;
-        int cumulativeUncompressedBlockLength = 0;
-
-        while (totalDecompressedCount < uncompressedSize) {
-            if (totalDecompressedCount == cumulativeUncompressedBlockLength) {
-                cumulativeUncompressedBlockLength += Integer.reverseBytes(input.getInt(inputOffset));
-                inputOffset += SIZE_OF_INT;
-            }
-            int compressedChunkLength = Integer.reverseBytes(input.getInt(inputOffset));
-            inputOffset += SIZE_OF_INT;
-            int decompressionSize = decompress(lzoDecompressor, input, inputOffset, compressedChunkLength, output, outputOffset);
-            totalDecompressedCount += decompressionSize;
-            outputOffset += decompressionSize;
-            inputOffset += compressedChunkLength;
-        }
-        checkArgument(outputOffset == uncompressedSize);
-        return wrappedBuffer(output, 0, uncompressedSize);
+        final BytesInput decompressed;
+        LzoCodec codec = new LzoCodec();
+        codec.setConf(new Configuration());
+        org.apache.hadoop.io.compress.Decompressor decompressor = codec.createDecompressor();
+
+        // input data
+        byte[] byteArray = (byte[]) input.getBase();
+        int inputOffset = (int) input.getAddress() - ARRAY_BYTE_BASE_OFFSET;
+        ByteArrayBytesInput bytes = new ByteArrayBytesInput(byteArray, inputOffset, input.length());
+
+        decompressor.reset();
+        InputStream is = codec.createInputStream(toInputStream(bytes), decompressor);
```

Comment:
Since you are using the same codec to createInputStream as you create the decompressor, you can just use `in = codec.createInputStream(in)` and no need to pre-create the decompressor.

Reply:
I think the initialization (for some reason) is necessary. If I don't set the conf in the decompressor, it will report errors: 
```java.lang.AssertionError: Configuration cannot be null!
	at org.anarres.lzo.hadoop.codec.LzoCodec.getDecompressionStrategy(LzoCodec.java:69)
	at org.anarres.lzo.hadoop.codec.LzoCodec.createDecompressor(LzoCodec.java:182)
	at org.anarres.lzo.hadoop.codec.LzoCodec.createInputStream(LzoCodec.java:163)
	at com.facebook.presto.hive.parquet.ParquetCompressionUtils.decompressLZO(ParquetCompressionUtils.java:99)
	at com.facebook.presto.hive.parquet.ParquetCompressionUtils.decompress(ParquetCompressionUtils.java:58)
	at com.facebook.presto.hive.parquet.reader.ParquetPageReader.readPage(ParquetPageReader.java:65)
	at com.facebook.presto.hive.parquet.reader.ParquetColumnReader.readNextPage(ParquetColumnReader.java:248)
	at com.facebook.presto.hive.parquet.reader.ParquetColumnReader.readPrimitive(ParquetColumnReader.java:158)
	at com.facebook.presto.hive.parquet.reader.ParquetReader.readPrimitive(ParquetReader.java:264)
	at com.facebook.presto.hive.parquet.reader.ParquetReader.readPrimitive(ParquetReader.java:246)
	at com.facebook.presto.hive.parquet.ParquetPageSource$ParquetBlockLoader.load(ParquetPageSource.java:277)
	at com.facebook.presto.hive.parquet.ParquetPageSource$ParquetBlockLoader.load(ParquetPageSource.java:253)
	at com.facebook.presto.spi.block.LazyBlock.assureLoaded(LazyBlock.java:255)
	at com.facebook.presto.spi.Page.assureLoaded(Page.java:241)
	at com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:265)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:379)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:278)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:645)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:272)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:975)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:492)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)```

Another way to avoid the initialization is to create a CodecFactory so that we can reuse the Decompressor object.